src/compiler/scala/tools/util/PathResolver.scala

```diff
@@ -268,9 +269,20 @@ abstract class PathResolverBase[BaseClassPathType <: ClassFileLookup[AbstractFil
 
     import classPathFactory._
 
+    private def javaBootClasspath = {
+      val cp = classesInPath(javaBootClassPath)
+      val okay = cp.exists(_.findClass("java/lang/Object.class").isDefined)
```

Comment:
can we avoid this scan unless we're under java 9?

Reply:
done